src/recorder/jamcontroller.cpp

```diff
@@ -169,6 +174,22 @@ void CJamController::SetRecordingDir ( QString newRecordingDir,
     }
 }
 
+void CJamController::OnRecordingFailed ( QString error )
+{
+    if ( !bEnableRecording )
+    {
+        // Recording has already been stopped, possibly
+        // by a previous OnRecordingFailed call from another client/thread.
+        return;
+    }
+    strRecorderErrMsg = error;
```

Comment:
We're single-threaded here, right?  So no need for mutex protection?

Reply:
Both `bEnableRecording` and `strRecorderErrMsg` access is similar to other recorder code, so I think it should be fine.

As far as I understand multithreading in Jamulus there are two relevant places:
- Mixing may be multithreaded (`-T`). This shouldn't affect any of those variables.
- The recorder uses its own thread. `pJamRecorder` lives in that thread, but is not able to touch these variables (which are only used in `pJamController` which should live on the main thread).